x/gamm/keeper/pool.go

```diff
@@ -103,6 +148,36 @@ func (k Keeper) setPool(ctx sdk.Context, pool types.PoolI) error {
 	return nil
 }
 
+// incrementPoolCount incrementes pool count by 1.
+func (k Keeper) incrementPoolCount(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+	poolCount := &gogotypes.UInt64Value{}
+	osmoutils.MustGet(store, types.KeyGammPoolCount, poolCount)
+	poolCount.Value = poolCount.Value + 1
+	osmoutils.MustSet(store, types.KeyGammPoolCount, poolCount)
+}
+
+// initializePoolCount initializes pool count to 0.
+func (k Keeper) initializePoolCount(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+	poolCount := &gogotypes.UInt64Value{Value: 0}
+	osmoutils.MustSet(store, types.KeyGammPoolCount, poolCount)
+}
+
```

Comment:
Sure. It might be useful to know the number of pools in `gamm`, in `concentrated-liquidity and in total in `swaprouter`. This is what updates the counter in gamm.